projects/storefrontlib/src/layout/a11y/keyboard-focus/visible/visible-focus.directive.ts

```diff
@@ -23,13 +23,29 @@ export class VisibleFocusDirective extends BaseFocusDirective {
     }
   }
 
-  @HostListener('keydown') handleKeydown() {
+  @HostListener('keydown', ['$event']) handleKeydown(event: KeyboardEvent) {
     if (this.shouldFocusVisible) {
-      this.mouseFocus = false;
+      this.mouseFocus = !this.isMouseEvent(event);
     }
   }
 
   protected get shouldFocusVisible(): boolean {
     return this.config?.disableMouseFocus;
   }
+
+  /**
+   * Indicates whether the current event is driven by the mouse or the keyboard.
+   *
+   * If the keyboard is used to navigate the OS or the browser, or used ot fill in a form,
+   * we consider the event to be a mouse event.
+   */
+  protected isMouseEvent(event: KeyboardEvent): boolean {
```

Comment:
IMO it can be a bit misleading to treat and name a 'mouse event' the subset of the exceptional keyboard events.

You know the family of the focus directives best. What do you think about adding another host binding and CSS class just for exceptional keyboard events? For example `.keyboard-focus-hidden`. 
As an alternative, we could deprecate `mouse`-related nomenclature in favour of `visible-focus`, i.e. `.no-visible-focus` CSS class and `disableVisibleFocus` config property instead of `disableMouseFocus`?

Reply:
Agreed. I'm not fond of yet another binding, but like the name improvement. I'll deprecate as you suggest and introduce a better name.